api/api.py

```diff
@@ -334,6 +347,53 @@ def historical(self, request, pk=None):
     def teste(self, request, pk=None):
         return Response("ola teste")
 
+    @action(detail=False, methods=['GET', 'POST'])
+    def expenses_and_revenues(self, request, pk=None):
+        params = ExpenseAndRevenueCountryIdFilterSerializer(data=request.GET)
+        params.is_valid(raise_exception=True)
+        country = params.validated_data.get('country')
+        budgets = Budget.objects.filter(country=country).order_by("year")
+        expensesRevenues = []
+
+        for budget in budgets:
+            budget_summary = BudgetSummary.objects.get(budget=budget)
+            if not (budget_summary.expense_functional_budget is None):
```

Comment:
You can simplify logic with python:
eg.: `if not (budget_summary.expense_functional_budget is None)` => `if budget_summary.expense_functional_budget`